pallets/subtensor/src/block_step.rs

```diff
@@ -82,6 +84,38 @@ impl<T: Config> Pallet<T> {
         LoadedEmission::<T>::get(netuid).unwrap()
     }
 
+    // Checks the current block emission and
+    // updates the block emission if the recalculated value is lower
+    //
+    pub fn check_halving() {
+        const ORIGINAL_EMISSION: f64 = 1_000_000_000.0; //Rao => 1 Tao
```

Comment:
I'm surprised that this isn't already defined somewhere. In fact, it might not be a good idea to define it here, and instead should be a config item where the runtime gets to define what the original emission is. We'd then be able to reuse it here via `T::ORIGINAL_EMISSION`.

Reply:
It is already defined in https://github.com/opentensor/subtensor/blob/09e0270028d3cb616aa781df80eae3467f8e6f9b/pallets/subtensor/src/lib.rs#L211

You should take advantage of that.